lib/svg-sprite/shape.js

```diff
@@ -513,20 +511,60 @@ SVGShape.prototype._determineDimensions = function(cb) {
 	// If the viewBox attribute didn't suffice: Render the SVG image
 	if (!this.width || !this.height) {
 		var that				= this;
-	    execFile(phantomjs, [dimensionsPhantomScript, this.getSVG(false), 'file://' + this.source.path], function (err, stdout, stderr) {
-	        if (err) {
-	            cb(err);
-	        } else if (stdout.length > 0) { // PhantomJS always outputs to stdout.
-	        	var dimensions	= JSON.parse(stdout.toString().trim());
-	        	that.width		= that._round(dimensions.width);
-	        	that.height		= that._round(dimensions.height);
-	        	cb(null);
-	        } else if (stderr.length > 0) {
-	            cb(new Error(stderr.toString().trim()));
-	        } else {
-	            cb(new Error('PhantomJS didn\'t return dimensions for "' + that.name + '"'));
-	        }
-	    });
+		var _browser, _page, _nodeHandle;
+		try {
+			puppeteer.launch()
+			.then(function(browser) {
+				_browser = browser;
+				return browser.newPage();
+			})
+			.then(function(page) {
+				_page = page;
+				return _page.setViewport({
+					width: 1280,
+					height: 1024
+				});
+			})
+			.then(function() {
+				return _page.goto('file://dummy.svg');
+			})
+			.then(function() {
+				var svg = that.getSVG(false);
+				return _page.setContent(
+					//'<svg xmlns="http://www.w3.org/2000/svg">' + 
+					'<?xml version="1.0" encoding="UTF-8"?>' + 
+					'<svg ' + svg.substr(svg.toLowerCase().indexOf('<svg') + 4) + '</svg>',
+					{ waitUntil: 'networkidle0' }
+				);
+			})
+			.then(function() {
+				return _page.$('svg');
+			})
+			.then(function(nodeHandle) {
+				_nodeHandle = nodeHandle;
+				return _page.evaluate(function(node) {
+					var rect = node.getBoundingClientRect();
+					return { width: rect.width, height: rect.height };
+				}, nodeHandle);
+			})
+			.then(function(bounds) {
+				that.width = that._round(bounds.width);
+				that.height = that._round(bounds.height);
+				return _nodeHandle.dispose();
+			})
```

Comment:
I think can be simplify the 3 function into 2 and eliminate the `_nodeHandle` variable.
```suggestion
			.then(function() {
				return _page.evaluate(function() {
					const rect = $('svg').getBoundingClientRect()
					return { width: rect.width, height: rect.height };
				});
			})
			.then(function(bounds) {
				that.width = that._round(bounds.width);
				that.height = that._round(bounds.height);
			})
```